synapse/handlers/sliding_sync.py

```diff
@@ -619,8 +629,19 @@ async def handle_room(room_id: str) -> None:
             sync_config=sync_config, to_token=to_token
         )
 
-        # TODO: Update this when we implement per-connection state
-        connection_token = 0
+        if has_lists or has_room_subscriptions:
+            connection_token = await self.connection_store.record_rooms(
+                sync_config=sync_config,
+                from_token=from_token,
+                sent_room_ids=relevant_room_map.keys(),
+                # TODO: We need to calculate which rooms have had updates since the `from_token` but were not included in the `sent_room_ids`
+                unsent_room_ids=[],
+            )
+        elif from_token:
+            connection_token = from_token.connection
+        else:
+            # Initial sync without a `from_token` starts a `0`
```

Comment:
My own typo from the [suggestion](https://github.com/element-hq/synapse/pull/17447#discussion_r1681421844):

```suggestion
            # Initial sync without a `from_token` starts at `0`
```